library/src/tests/resources/src/durrhoyerlibrary.qs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Test {
+    import Std.Math.*;
+    import Std.Arrays.*;
+    import Std.Convert.*;
+    import Std.Random.*
+    import Std.DurrHoyerLibrary.*;
+
+    // Function to find the maximum element in an array
+    function MaxIntArray(arr : Int[]) : Int {
+        mutable max = arr[0];
+        for i in arr[1..Length(arr) - 1] {
+            if (arr[i] > max) {
+                set max = arr[i];
+            }
+        }
+        return max;
+    }
+
+    // Common function to compute the probability of finding the minimum or maximum index
+    operation ComputeDurrHoyerProbabilityWithShots(
+        shots : Int,
+        testLists : Int[][],
+        expectedIndices : Int[],
+        operationType : String
+    ) : Unit {
+        for (list, expectedIndex) in Zipped(testLists, expectedIndices) {
+            let lengthList : Int = Length(list);
+            let candidateIndex : Int = DrawRandomInt(0, lengthList - 1);
+            let maxValue = MaxIntArray(list);
+            let double : Double = IntAsDouble(maxValue + 1);
+            let log : Double = Log(double) / Log(2.0);
+            let nQubits = Ceiling(log);
+            let testLists : Int[][] = [
```

Comment:
Some comments on the tests:
- What happens if some/all of the values are negative in a `testList`?
- What happens if 0 is passed as input  in a `testList`?
- Can we please rearrange the values so that expectedMaxIndex is not 0 in all the test cases?

Reply:
for negatives in ConvertToBinary:

if (value < 0 or value > maxVal) {
    fail $"Value {value} cannot be represented with {length} bits.";
}

ConvertToBinary will fail if there are negatives

Reply:
Added unit test for 0 in input list